File: impl/app/android/app/src/main/java/com/ak/contexta/ui/onboarding/OnboardingScreen.kt
package com.ak.contexta.ui.onboarding

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.MenuBook
import androidx.compose.material.icons.outlined.AutoStories
import androidx.compose.material.icons.outlined.School
import androidx.compose.material.icons.outlined.WorkspacePremium
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import com.ak.contexta.ui.theme.Background
import com.ak.contexta.ui.theme.Hairline
import com.ak.contexta.ui.theme.Ink
import com.ak.contexta.ui.theme.Muted
import com.ak.contexta.ui.theme.MutedSoft
import com.ak.contexta.ui.theme.OnPrimary
import com.ak.contexta.ui.theme.Primary
import com.ak.contexta.ui.theme.PrimaryDisabled
import com.ak.contexta.ui.theme.SurfaceCard
import com.ak.contexta.ui.theme.SurfaceStrong

@Composable
fun OnboardingScreen(
    onComplete: () -> Unit,
    viewModel: OnboardingViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsState()

    // Auto-redirect to Home if user already completed onboarding
    LaunchedEffect(Unit) {
        if (viewModel.isAlreadyOnboarded()) {
            onComplete()
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Background)
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 24.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(40.dp))

        // Logo area
        Text(
            text = "Contexta",
            style = MaterialTheme.typography.displayLarge
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = "语境 · 沉浸式英语阅读",
            style = MaterialTheme.typography.bodyMedium,
            color = Muted
        )

        Spacer(modifier = Modifier.height(32.dp))

        // Step content
        when (state.currentStep) {
            1 -> Step1Level(
                selectedLevel = state.selectedLevel,
                onSelectLevel = { viewModel.selectLevel(it) }
            )
            2 -> Step2DailyCount(
                selectedCount = state.selectedDailyCount,
                onSelectCount = { viewModel.selectDailyCount(it) }
            )
            3 -> Step3Confirmation(
                level = state.selectedLevel ?: "",
                dailyCount = state.selectedDailyCount ?: 0
            )
        }

        Spacer(modifier = Modifier.weight(1f))

        // Progress dots
        ProgressDots(currentStep = state.currentStep)

        Spacer(modifier = Modifier.height(16.dp))

        // Navigation buttons
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            if (state.currentStep > 1) {
                OutlinedButton(
                    onClick = { viewModel.previousStep() },
                    modifier = Modifier.weight(1f),
                    shape = RoundedCornerShape(8.dp),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = Ink
                    )
                ) {
                    Text("上一步")
                }
            }

            val canProceed = when (state.currentStep) {
                1 -> state.selectedLevel != null
                2 -> state.selectedDailyCount != null
                3 -> true
                else -> false
            }

            Button(
                onClick = {
                    if (state.currentStep < 3) {
                        viewModel.nextStep()
                    } else {
                        viewModel.completeOnboarding(onComplete)
                    }
                },
                enabled = canProceed,
                modifier = Modifier.weight(1f),
                shape = RoundedCornerShape(8.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Primary,
                    contentColor = OnPrimary,
                    disabledContainerColor = PrimaryDisabled.copy(alpha = 0.5f),
                    disabledContentColor = MutedSoft
                )
            ) {
                Text(if (state.currentStep < 3) "下一步" else "开始学习")
            }
        }

        Spacer(modifier = Modifier.height(32.dp))
    }
}

@Composable
private fun Step1Level(
    selectedLevel: String?,
    onSelectLevel: (String) -> Unit
) {
    Column {
        Text(
            text = "选择你的英文水平",
            style = MaterialTheme.typography.displayMedium
        )
        Spacer(modifier = Modifier.height(24.dp))

        LevelOption(
            value = "LOW",
            label = "初级 · LOW",
            desc = "小学、初中水平，从基础开始",
            isSelected = selectedLevel == "LOW",
            onClick = { onSelectLevel("LOW") },
            icon = Icons.Outlined.School
        )
        LevelOption(
            value = "MEDIUM",
            label = "中级 · MEDIUM",
            desc = "高中、大学四六级水平",
            isSelected = selectedLevel == "MEDIUM",
            onClick = { onSelectLevel("MEDIUM") },
            icon = Icons.Outlined.AutoStories
        )
        LevelOption(
            value = "HIGH",
            label = "高级 · HIGH",
            desc = "专八、托福、雅思水平",
            isSelected = selectedLevel == "HIGH",
            onClick = { onSelectLevel("HIGH") },
            icon = Icons.Outlined.WorkspacePremium
        )
    }
}

@Composable
private fun Step2DailyCount(
    selectedCount: Int?,
    onSelectCount: (Int) -> Unit
) {
    Column {
        Text(
            text = "每日文章数量",
            style = MaterialTheme.typography.displayMedium
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = "每天为你生成几篇文章？",
            style = MaterialTheme.typography.bodyMedium,
            color = Muted
        )
        Spacer(modifier = Modifier.height(20.dp))

        CountOption(
            count = 1,
            label = "1 篇",
            desc = "轻松起步",
            isSelected = selectedCount == 1,
            onClick = { onSelectCount(1) }
        )
        CountOption(
            count = 3,
            label = "3 篇",
            desc = "适中节奏",
            isSelected = selectedCount == 3,
            onClick = { onSelectCount(3) }
        )
        CountOption(
            count = 5,
            label = "5 篇",
            desc = "充分练习",
            isSelected = selectedCount == 5,
            onClick = { onSelectCount(5) }
        )
    }
}

@Composable
private fun Step3Confirmation(
    level: String,
    dailyCount: Int
) {
    Column {
        Text(
            text = "准备好了！",
            style = MaterialTheme.typography.displayMedium
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = "系统将根据你的设置，每天推送匹配水平的英文文章。首次生成需要一些时间，请稍候。",
            style = MaterialTheme.typography.bodyMedium,
            color = Muted
        )
        Spacer(modifier = Modifier.height(24.dp))

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(12.dp))
                .background(SurfaceCard)
                .padding(20.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Icon(
                imageVector = Icons.AutoMirrored.Outlined.MenuBook,
                contentDescription = null,
                tint = Primary,
                modifier = Modifier.size(40.dp)
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "你的专属阅读之旅即将开始",
                style = MaterialTheme.typography.titleMedium
            )
            Spacer(modifier = Modifier.height(4.dp))
            Text(
                text = "每天坚持阅读，不知不觉提升英语",
                style = MaterialTheme.typography.bodySmall,
                color = Muted
            )
            Spacer(modifier = Modifier.height(16.dp))
            Text(
                text = "水平：${levelLabel(level)}   |   每日 ${dailyCount} 篇",
                style = MaterialTheme.typography.bodyMedium,
                color = Primary,
                textAlign = TextAlign.Center
            )
        }
    }
}

@Composable
private fun LevelOption(
    value: String,
    label: String,
    desc: String,
    isSelected: Boolean,
    onClick: () -> Unit,
    icon: ImageVector
) {
    val borderColor = if (isSelected) Primary else Hairline
    val bgColor = if (isSelected) SurfaceStrong else SurfaceCard

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(bgColor)
            .border(2.dp, borderColor, RoundedCornerShape(12.dp))
            .clickable { onClick() }
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = if (isSelected) Primary else MutedSoft,
            modifier = Modifier.size(28.dp)
        )
        Spacer(modifier = Modifier.width(10.dp))
        Box(
            modifier = Modifier
                .size(22.dp)
                .clip(CircleShape)
                .border(2.dp, if (isSelected) Primary else Hairline, CircleShape),
            contentAlignment = Alignment.Center
        ) {
            if (isSelected) {
                Box(
                    modifier = Modifier
                        .size(12.dp)
                        .clip(CircleShape)
                        .background(Primary)
                )
            }
        }
        Spacer(modifier = Modifier.width(14.dp))
        Column {
            Text(text = label, style = MaterialTheme.typography.titleMedium)
            Text(
                text = desc,
                style = MaterialTheme.typography.bodySmall,
                color = Muted
            )
        }
    }
}

@Composable
private fun CountOption(
    count: Int,
    label: String,
    desc: String,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    val borderColor = if (isSelected) Primary else Hairline
    val bgColor = if (isSelected) SurfaceStrong else SurfaceCard

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(bgColor)
            .border(2.dp, borderColor, RoundedCornerShape(12.dp))
            .clickable { onClick() }
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(22.dp)
                .clip(CircleShape)
                .border(2.dp, if (isSelected) Primary else Hairline, CircleShape),
            contentAlignment = Alignment.Center
        ) {
            if (isSelected) {
                Box(
                    modifier = Modifier
                        .size(12.dp)
                        .clip(CircleShape)
                        .background(Primary)
                )
            }
        }
        Spacer(modifier = Modifier.width(14.dp))
        Column {
            Text(text = label, style = MaterialTheme.typography.titleMedium)
            Text(
                text = desc,
                style = MaterialTheme.typography.bodySmall,
                color = Muted
            )
        }
    }
}

@Composable
private fun ProgressDots(currentStep: Int) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        repeat(3) { index ->
            val step = index + 1
            val isActive = step == currentStep
            val isDone = step < currentStep

            Box(
                modifier = Modifier
                    .height(8.dp)
                    .then(
                        if (isActive) Modifier.width(24.dp)
                        else Modifier.size(8.dp)
                    )
                    .clip(if (isActive) RoundedCornerShape(4.dp) else CircleShape)
                    .background(
                        when {
                            isActive -> Primary
                            isDone -> Primary.copy(alpha = 0.4f)
                            else -> Hairline
                        }
                    )
            )
        }
    }
}

private fun levelLabel(level: String): String = when (level) {
    "LOW" -> "初级"
    "MEDIUM" -> "中级"
    "HIGH" -> "高级"
    else -> level
}

File: impl/app/flutter/lib/ui/onboarding/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/components/app_button.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/app_dimens.dart';
import '../../core/theme/app_type.dart';
import 'onboarding_controller.dart';

/// Onboarding 引导页（对照 Kotlin OnboardingScreen.kt）：
/// 3 步向导（水平 → 每日篇数 → 确认），底部进度点 + 上一步/下一步按钮。
class OnboardingScreen extends ConsumerWidget {
  const OnboardingScreen({super.key, this.onComplete});

  /// 完成引导后的跳转回调（由路由层注入：context.go('/home')）。
  final VoidCallback? onComplete;

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final state = ref.watch(onboardingControllerProvider);

    return Scaffold(
      backgroundColor: AppColors.background,
      body: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            // 品牌区
            const SizedBox(height: 40),
            Text(
              'Contexta',
              textAlign: TextAlign.center,
              style: AppType.textTheme.displayLarge,
            ),
            const SizedBox(height: 4),
            Text(
              '语境 · 沉浸式英语阅读',
              textAlign: TextAlign.center,
              style: AppType.textTheme.bodyMedium
                  ?.copyWith(color: AppColors.muted),
            ),
            const SizedBox(height: 32),

            // 步骤内容
            Expanded(
              child: SingleChildScrollView(
                padding: const EdgeInsets.symmetric(horizontal: 24),
                child: switch (state.currentStep) {
                  1 => _Step1Level(
                      selectedLevel: state.selectedLevel,
                      onSelectLevel: ref
                          .read(onboardingControllerProvider.notifier)
                          .selectLevel,
                    ),
                  2 => _Step2DailyCount(
                      selectedDailyCount: state.selectedDailyCount,
                      onSelectCount: ref
                          .read(onboardingControllerProvider.notifier)
                          .selectDailyCount,
                    ),
                  _ => _Step3Confirmation(
                      level: state.selectedLevel ?? '',
                      dailyCount: state.selectedDailyCount ?? 0,
                    ),
                },
              ),
            ),

            // 底部操作区
            Column(
              children: [
                _ProgressDots(currentStep: state.currentStep),
                const SizedBox(height: 16),
                Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 24),
                  child: Row(
                    children: [
                      if (state.currentStep > 1) ...[
                        Expanded(
                          child: AppButton(
                            text: '上一步',
                            variant: AppButtonVariant.secondary,
                            onClick: ref
                                .read(onboardingControllerProvider.notifier)
                                .previousStep,
                          ),
                        ),
                        const SizedBox(width: 12),
                      ],
                      Expanded(
                        child: AppButton(
                          text: state.currentStep < 3 ? '下一步' : '开始学习',
                          enabled: state.canProceed,
                          onClick: () {
                            final controller =
                                ref.read(onboardingControllerProvider.notifier);
                            if (state.currentStep < 3) {
                              controller.nextStep();
                            } else {
                              controller.completeOnboarding(
                                  onComplete ?? () {});
                            }
                          },
                        ),
                      ),
                    ],
                  ),
                ),
                const SizedBox(height: 32),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

/// 第 1 步：选择英文水平。
class _Step1Level extends StatelessWidget {
  const _Step1Level({required this.selectedLevel, required this.onSelectLevel});

  final String? selectedLevel;
  final ValueChanged<String> onSelectLevel;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text('选择你的英文水平', style: AppType.textTheme.displayMedium),
        const SizedBox(height: 24),
        _LevelOption(
          label: '初级 · LOW',
          desc: '小学、初中水平，从基础开始',
          isSelected: selectedLevel == 'LOW',
          onClick: () => onSelectLevel('LOW'),
          icon: Icons.school_outlined,
        ),
        _LevelOption(
          label: '中级 · MEDIUM',
          desc: '高中、大学四六级水平',
          isSelected: selectedLevel == 'MEDIUM',
          onClick: () => onSelectLevel('MEDIUM'),
          icon: Icons.auto_stories_outlined,
        ),
        _LevelOption(
          label: '高级 · HIGH',
          desc: '专八、托福、雅思水平',
          isSelected: selectedLevel == 'HIGH',
          onClick: () => onSelectLevel('HIGH'),
          icon: Icons.workspace_premium_outlined,
        ),
      ],
    );
  }
}

/// 第 2 步：每日文章数量。
class _Step2DailyCount extends StatelessWidget {
  const _Step2DailyCount({
    required this.selectedDailyCount,
    required this.onSelectCount,
  });

  final int? selectedDailyCount;
  final ValueChanged<int> onSelectCount;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text('每日文章数量', style: AppType.textTheme.displayMedium),
        const SizedBox(height: 4),
        Text(
          '每天为你生成几篇文章？',
          style: AppType.textTheme.bodyMedium?.copyWith(color: AppColors.muted),
        ),
        const SizedBox(height: 20),
        _CountOption(
          count: 1,
          label: '1 篇',
          desc: '轻松起步',
          isSelected: selectedDailyCount == 1,
          onClick: () => onSelectCount(1),
        ),
        _CountOption(
          count: 3,
          label: '3 篇',
          desc: '适中节奏',
          isSelected: selectedDailyCount == 3,
          onClick: () => onSelectCount(3),
        ),
        _CountOption(
          count: 5,
          label: '5 篇',
          desc: '充分练习',
          isSelected: selectedDailyCount == 5,
          onClick: () => onSelectCount(5),
        ),
      ],
    );
  }
}

/// 第 3 步：确认页。
class _Step3Confirmation extends StatelessWidget {
  const _Step3Confirmation({required this.level, required this.dailyCount});

  final String level;
  final int dailyCount;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text('准备好了！', style: AppType.textTheme.displayMedium),
        const SizedBox(height: 4),
        Text(
          '系统将根据你的设置，每天推送匹配水平的英文文章。首次生成需要一些时间，请稍候。',
          style: AppType.textTheme.bodyMedium?.copyWith(color: AppColors.muted),
        ),
        const SizedBox(height: 24),
        Container(
          padding: const EdgeInsets.all(20),
          decoration: BoxDecoration(
            color: AppColors.surfaceCard,
            borderRadius: BorderRadius.circular(AppRadius.md),
          ),
          child: Column(
            children: [
              Icon(Icons.menu_book_outlined,
                  size: 40, color: AppColors.primary),
              const SizedBox(height: 8),
              Text(
                '你的专属阅读之旅即将开始',
                style: AppType.textTheme.titleMedium,
              ),
              const SizedBox(height: 4),
              Text(
                '每天坚持阅读，不知不觉提升英语',
                style: AppType.textTheme.bodySmall
                    ?.copyWith(color: AppColors.muted),
              ),
              const SizedBox(height: 16),
              Text(
                '水平：${_levelLabel(level)}   |   每日 $dailyCount 篇',
                textAlign: TextAlign.center,
                style: AppType.textTheme.bodyMedium
                    ?.copyWith(color: AppColors.primary),
              ),
            ],
          ),
        ),
      ],
    );
  }
}

/// 水平/篇数选择卡片：选中 2dp Primary 边框 + SurfaceStrong 底 +
/// 22dp 圆形单选指示；未选中 Hairline 边框 + SurfaceCard 底。
class _OptionCard extends StatelessWidget {
  const _OptionCard({
    required this.label,
    required this.desc,
    required this.isSelected,
    required this.onClick,
    this.icon,
  });

  final String label;
  final String desc;
  final bool isSelected;
  final VoidCallback onClick;
  final IconData? icon;

  @override
  Widget build(BuildContext context) {
    final borderColor = isSelected ? AppColors.primary : AppColors.hairline;
    final bgColor = isSelected ? AppColors.surfaceStrong : AppColors.surfaceCard;

    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Material(
        color: bgColor,
        borderRadius: BorderRadius.circular(AppRadius.md),
        child: InkWell(
          onTap: onClick,
          borderRadius: BorderRadius.circular(AppRadius.md),
          child: Container(
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(
              borderRadius: BorderRadius.circular(AppRadius.md),
              border: Border.all(color: borderColor, width: 2),
            ),
            child: Row(
              children: [
                if (icon != null) ...[
                  Icon(icon, size: 28,
                      color: isSelected ? AppColors.primary : AppColors.mutedSoft),
                  const SizedBox(width: 10),
                ],
                Container(
                  width: 22,
                  height: 22,
                  decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    border: Border.all(
                      color: isSelected ? AppColors.primary : AppColors.hairline,
                      width: 2,
                    ),
                  ),
                  child: isSelected
                      ? Center(
                          child: Container(
                            width: 12,
                            height: 12,
                            decoration: const BoxDecoration(
                              shape: BoxShape.circle,
                              color: AppColors.primary,
                            ),
                          ),
                        )
                      : null,
                ),
                const SizedBox(width: 14),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(label, style: AppType.textTheme.titleMedium),
                      Text(
                        desc,
                        style: AppType.textTheme.bodySmall
                            ?.copyWith(color: AppColors.muted),
                      ),
                    ],
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

class _LevelOption extends StatelessWidget {
  const _LevelOption({
    required this.label,
    required this.desc,
    required this.isSelected,
    required this.onClick,
    required this.icon,
  });

  final String label;
  final String desc;
  final bool isSelected;
  final VoidCallback onClick;
  final IconData icon;

  @override
  Widget build(BuildContext context) => _OptionCard(
        label: label,
        desc: desc,
        isSelected: isSelected,
        onClick: onClick,
        icon: icon,
      );
}

class _CountOption extends StatelessWidget {
  const _CountOption({
    required this.count,
    required this.label,
    required this.desc,
    required this.isSelected,
    required this.onClick,
  });

  final int count;
  final String label;
  final String desc;
  final bool isSelected;
  final VoidCallback onClick;

  @override
  Widget build(BuildContext context) => _OptionCard(
        label: label,
        desc: desc,
        isSelected: isSelected,
        onClick: onClick,
      );
}

/// 3 点进度条：当前步 Primary 胶囊（24x8），已完成 Primary 40%，未到 Hairline。
class _ProgressDots extends StatelessWidget {
  const _ProgressDots({required this.currentStep});

  final int currentStep;

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        for (var i = 1; i <= 3; i++) ...[
          if (i > 1) const SizedBox(width: 8),
          Container(
            width: i == currentStep ? 24 : 8,
            height: 8,
            decoration: BoxDecoration(
              color: switch ((i, i < currentStep)) {
                (_, true) => AppColors.primary.withValues(alpha: 0.4),
                (_, false) when i == currentStep => AppColors.primary,
                _ => AppColors.hairline,
              },
              borderRadius: BorderRadius.circular(
                  i == currentStep ? 4 : 999),
            ),
          ),
        ],
      ],
    );
  }
}

String _levelLabel(String level) => switch (level) {
      'LOW' => '初级',
      'MEDIUM' => '中级',
      'HIGH' => '高级',
      _ => level,
    };
